src/Files.App/UserControls/SidebarControl.xaml.cs

```diff
@@ -360,129 +356,71 @@ private void HideSection()
 
 		private async void OpenInNewPane()
 		{
-			if (await CheckEmptyDrive((rightClickedItem as INavigationControlItem)?.Path))
-			{
+			if (await CheckEmptyDrive((rightClickedItem as INavigationControlItem).Path))
 				return;
-			}
+
 			SidebarItemNewPaneInvoked?.Invoke(this, new SidebarItemNewPaneInvokedEventArgs(rightClickedItem));
 		}
 
 		private async void OpenInNewTab()
 		{
 			if (await CheckEmptyDrive(rightClickedItem.Path))
-			{
 				return;
-			}
+
 			await NavigationHelpers.OpenPathInNewTab(rightClickedItem.Path);
 		}
 
 		private async void OpenInNewWindow()
 		{
 			if (await CheckEmptyDrive(rightClickedItem.Path))
-			{
 				return;
-			}
+
 			await NavigationHelpers.OpenPathInNewWindowAsync(rightClickedItem.Path);
 		}
 
 		private void PinItem()
 		{
 			if (rightClickedItem is DriveItem)
-			{
 				App.SidebarPinnedController.Model.AddItem(rightClickedItem.Path);
-			}
 		}
-
 		private void UnpinItem()
 		{
 			if (rightClickedItem.Section == SectionType.Favorites || rightClickedItem is DriveItem)
-			{
 				App.SidebarPinnedController.Model.RemoveItem(rightClickedItem.Path);
-			}
 		}
 
 		private void MoveItemToTop()
 		{
-			if (rightClickedItem.Section == SectionType.Favorites)
-			{
-				bool isSelectedSidebarItem = false;
-
-				if (SelectedSidebarItem == rightClickedItem)
-				{
-					isSelectedSidebarItem = true;
-				}
-
-				int oldIndex = App.SidebarPinnedController.Model.IndexOfItem(rightClickedItem);
-				App.SidebarPinnedController.Model.MoveItem(rightClickedItem, oldIndex, 0);
-
-				if (isSelectedSidebarItem)
-				{
-					SetValue(SelectedSidebarItemProperty, rightClickedItem);
-				}
-			}
+			MoveItemToNewIndex(0);
 		}
 
 		private void MoveItemUp()
 		{
-			if (rightClickedItem.Section == SectionType.Favorites)
-			{
-				bool isSelectedSidebarItem = false;
-
-				if (SelectedSidebarItem == rightClickedItem)
-				{
-					isSelectedSidebarItem = true;
-				}
-
-				int oldIndex = App.SidebarPinnedController.Model.IndexOfItem(rightClickedItem);
-				App.SidebarPinnedController.Model.MoveItem(rightClickedItem, oldIndex, oldIndex - 1);
-
-				if (isSelectedSidebarItem)
-				{
-					SetValue(SelectedSidebarItemProperty, rightClickedItem);
-				}
-			}
+			MoveItemToNewIndex(App.SidebarPinnedController.Model.IndexOfItem(rightClickedItem) - 1);
```

Comment:
Should this code be moved to the ViewModel? We can create a command to trigger the code.

Reply:
Don't think so, because the method interacts with the UI